src/js/component/analytics.js

```diff
@@ -0,0 +1,50 @@
+function Analytics ($module) {}
```

Comment:
If this is all working then maybe it's fine, but as Steve said, this is the odd line, as you're decalring `Analytics` as a function (which accepts a `$module` variable).

The simplest fix that might avoid this being confusing in the future could be to replace the line with:

`const Analytics = {}`

Which simply declares `Analytics` as an object, and then you are attaching your functions onto properites of the object, and should keep all your scope.

Or go with Steve's last suggestion and have all the functions staright in the file, remove all the `this`'s and export the functions you need available outside of the module.